test/integration/tcp_tunneling_integration_test.cc

```diff
@@ -429,20 +429,31 @@ TEST_P(TcpTunnelingIntegrationTest, CloseUpstreamFirst) {
   // Send data from upstream to downstream with an end stream and make sure the data is received
   // before the connection is half-closed.
   upstream_request_->encodeData(12, true);
```

Comment:
Do you think Envoy should do half close semantics when sending a CONNECT response?  If we did, the codec would frame with a half-close here and you wouldn't need the upstream close.

Reply:
yep, I think it could make sense, but I would tend to address it in a dedicated PR unless you consider this a blocker.